### ISSUE TITLE
Assertion fail in falconConsensus

### ISSUE THREAD

**Issue by Tierhon** [marbl/canu#727]
I get this error on one partition when correcting reads from a about 60x coverage PacBio (P6C4 and Sequel) dataset for a fish.
`Processing read 5101649 of length 16844 with 531 evidence reads.
falconsense: correction/falconConsensus.C:235: falconData* falconConsensus::getConsensus(uint32, alignTagList**, uint32): Assertion `g_best_score > 2.2250738585072014e-308' failed.
`
This is with version:
`-- Canu snapshot v1.6 +118 changes (r8530 87eeba06b16aefcaca80b6d4514bcdc831d795ab)
`
Running with these options:

`corMhapSensitivity=high \
corMinCoverage=0 \
corOutCoverage=200 \
correctedErrorRate=0.105 \`

Because of high heterozygosity (looks like indels several hundred bases every 10k or so). 

Can you see the issue?

Thank you.

Ole

**Comment by skoren** [comment]
We have not seen that crash before. The code in question underwent lots of changes since 1.6 and still has some issues remaining. I would recommend using the 1.6 release, unfortunately there isn't an easy way to use the old version with the new code due to other large-scale changes. I just committed an updated ```generateCorrectionLayouts``` file which will let you run the old version of falcon sense from 1.6 on the new data to generate a fasta file. You'd then have to join the fasta together manually as the corrected reads. Let us know if the old version works or has the same crash. You can run them as:
```
$bin/generateCorrectionLayouts -b $bgn -e $end \
  -rl ./asm.readsToCorrect \
  -G $gkpStore \
  -C ../asm.corStore \
  -eM 500 -eL 1000 \
&& \
  touch ./results/$jobid.dump.success \
) \
| \
$bin/falcon_sense \
  --min_idt 0.7 \
  --min_len 1000\
  --max_read_len 300000 \
  --min_ovl_len 500\
  --min_cov 0 \
  --n_core 4 \
  > ./results/$jobid.fasta.WORKING \
 2> ./results/$jobid.err \
&& \
mv ./results/$jobid.fasta.WORKING ./results/$jobid.fasta \
```

**Comment by Tierhon** [comment]
Hi Koren,
the old code did not crash in this instance. However, I forgot to add the fasta manually since I just submitted the job to the cluster. I am basically just searching for what works now. 1.6 is awfully slow in the steps around generateCorrectionLayouts, and head is much quicker. 

I will finish this assembly and assess the results, and likely wait for 1.7 before doing assembly on this species again.

**Comment by skoren** [comment]
Yes, the update in the tip is supposed to eliminate one pass of the overlap store and decrease the time for the correction of reads as well. Once you have the fasta from running the original version of falcon_sense, you can assemble with the tip as you would with any corrected read input.

**Comment by npbhavya** [comment]
Hi, 

I ran into a similar error when running canu on a human genome. Here is my command 
canu -p test-true -d test-true gridOptions="-l vmem=250gb,,walltime=20:00:00" useGrid=true genomeSize=5.8g minReadLength=300 minOverlapLength=150 -nanopore-raw INPUT.fa

The error is "Processing read 1 of length 430 with 58 evidence reads.
falconsense: correction/falconConsensus.C:235: falconData* falconConsensus::getConsensus(uint32, alignTagList**, uint32): Assertion `g_best_score > double(2.22507385850720138309e-308L)' failed."

canu --version is 
"Canu snapshot v1.6 +115 changes (r852725ec5a9d375d790a1268108b5de28dddc45f1085)" 

Do you know why this may still be happening? 
Thank You, 
Bhavya

**Comment by skoren** [comment]
Yes, it is not yet fixed so until it is I recommend using the 1.6 release for the assembly or at least for correction.

**Comment by brianwalenz** [comment]
Fixed!  (finally)
